### DISCUSSION TITLE
Getting error in production when using conversejs

### DISCUSSION THREAD

**Discussion by boristian** [react-static/react-static#1440] [General]
Hi,
at first, thanks for this awesome open source project. Great work 👍 

In a project i am react-static and [conversejs](https://github.com/conversejs/converse.js) together. In development mode everything works fine, but when in production i get the following error:

```sh
backbone.browserStorage.js:215 Uncaught TypeError: localforage.defineDriver is not a function
    at Module.<anonymous> (backbone.browserStorage.js:215)
    at __webpack_require__ (bootstrap:63)
    at Object.load (entry.js:60)
    at Object.initialize (entry.js:31)
    at Module.6 (main.7c8f7011.js:1)
    at a (main.7c8f7011.js:1)
    at Module.<anonymous> (main.7c8f7011.js:1)
    at Module.95 (main.7c8f7011.js:1)
    at a (main.7c8f7011.js:1)
    at Object.<anonymous> (vendors~main.3c766d42.js:1)
```

The error comes from the [localforage](https://github.com/localForage/localForage) package which conversejs depends on.

Any idea what might be the problem?

**Comment by SleeplessByte** [comment]
During build, the website is being "prerendered" (made static) in node.js. Here there is no access to `document` and `window`. 

You'll want to conditionally import `conversejs`, if you're in a browser environment, for example:

```js
if (typeof document !== undefined) {
  import('conversejs')
    .then((stuff) => {
      ... // use conversejs
    })
}
```

**Comment by boristian** [comment]
@SleeplessByte thanks for your answer. Im initializing conversejs already coinditionally

```js
const IS_BROWSER = typeof window !== 'undefined'

if (IS_BROWSER) {
  window.converse.initialize(window.__CONVERSEJS_CONFIG__)
}
```

The import is via script-tag and not conditional, maybe there is the problem.

**Reply by SleeplessByte** [reply]
That is correct. You can create the script tag in this block instead:

```javascript
const script = document.createElement('script')
script.src = ...
...
document.body.appendChild(script)

// You probably want to wait for the page to be loaded or the script 
// to be loaded before executing your initialize
```

**Reply by boristian** [reply]
@SleeplessByte thanks for your help. I loaded the script conditionally and initialized conversejs on load but sadly it did not help. The error is still the same, so i might have to search somewhere else. I really appreciate you taking your time for trying to help me. Thanks a lot!
